lib/buy/dfx/dfx_buy_provider.dart

```diff
@@ -135,8 +118,178 @@ class DFXBuyProvider extends BuyProvider {
+          return responseData.first as Map<String, dynamic>;
+        } else if (responseData is Map<String, dynamic>) {
+          return responseData;
+        } else {
+          log('DFX: Does not support this asset name : ${blockchain}');
+        }
+      } else {
+        log('DFX: Failed to fetch assets: ${response.statusCode}');
+      }
+    } catch (e) {
+      log('DFX: Error fetching assets: $e');
+    }
+    return {};
+  }
+
+  Future<List<PaymentMethod>> getAvailablePaymentTypes(
+      String fiatCurrency, String cryptoCurrency, bool isBuyAction) async {
+    final List<PaymentMethod> paymentMethods = [];
+
+    if (isBuyAction) {
+      final fiatBuyCredentials = await fetchFiatCredentials(fiatCurrency);
+      if (fiatBuyCredentials.isNotEmpty) {
+        fiatBuyCredentials.forEach((key, value) {
+          if (key == 'limits') {
+            final limits = value as Map<String, dynamic>;
+            limits.forEach((paymentMethodKey, paymentMethodValue) {
+              final min = _toDouble(paymentMethodValue['minVolume']);
+              final max = _toDouble(paymentMethodValue['maxVolume']);
+              if (min != null && max != null && min > 0 && max > 0) {
+                final paymentMethod = PaymentMethod.fromDFX(
+                    paymentMethodKey, _getPaymentTypeByString(paymentMethodKey));
+                paymentMethods.add(paymentMethod);
+              }
+            });
+          }
+        });
+      }
+    } else {
+      final assetCredentials = await fetchAssetCredential(cryptoCurrency);
+      if (assetCredentials.isNotEmpty) {
+        if (assetCredentials['sellable'] == true) {
+          final availablePaymentTypes = [
+            PaymentType.bankTransfer,
+            PaymentType.creditCard,
+            PaymentType.sepa
+          ];
+          availablePaymentTypes.forEach((element) {
+            final paymentMethod = PaymentMethod.fromDFX(normalizePaymentMethod(element)!, element);
+            paymentMethods.add(paymentMethod);
+          });
+        }
+      }
+    }
+
+    return paymentMethods;
+  }
+
   @override
-  Future<void> launchProvider(BuildContext context, bool? isBuyAction) async {
+  Future<List<Quote>?> fetchQuote(
+      {required CryptoCurrency cryptoCurrency,
+      required FiatCurrency fiatCurrency,
+      required double amount,
+      required bool isBuyAction,
+      required String walletAddress,
+      PaymentType? paymentType,
+      String? countryCode}) async {
+    /// if buying with any currency other than eur or chf then DFX is not supported
+
+    if (isBuyAction && (fiatCurrency != FiatCurrency.eur && fiatCurrency != FiatCurrency.chf)) {
+      return null;
+    }
+
+    String? paymentMethod;
+    if (paymentType != null && paymentType != PaymentType.all) {
+      paymentMethod = normalizePaymentMethod(paymentType);
+      if (paymentMethod == null) paymentMethod = paymentType.name;
+    } else {
+      paymentMethod = 'Bank';
+    }
```

Comment:
DFX requires a payment method to be set, while, for example, Onramper will return all available quotes if the payment method is null. Moreover, I set ‘Bank’ because DFX doesn’t support buying/selling XMR with a credit card, so users won’t see this provider initially for an XMR wallet.

Reply:
ok but I see this if condition in other providers as well, not just dfx, can you double check it please